pkg/controller/statefulset/stateful_set_status_updater.go

```diff
@@ -56,8 +56,7 @@ func (ssu *realStatefulSetStatusUpdater) UpdateStatefulSetStatus(
 	// don't wait due to limited number of clients, but backoff after the default number of steps
 	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
 		set.Status = *status
-		// TODO: This context.TODO should use a real context once we have RetryOnConflictWithContext
-		_, updateErr := ssu.client.AppsV1().StatefulSets(set.Namespace).UpdateStatus(context.TODO(), set, metav1.UpdateOptions{})
+		_, updateErr := ssu.client.AppsV1().StatefulSets(set.Namespace).UpdateStatus(ctx, set, metav1.UpdateOptions{})
```

Comment:
this is not solving the TODO

Reply:
@aojea does it really need a RetryOnConflictWithContext? I think we can use the outer ctx which passed from  `UpdateStatefulSetStatus` direclty

Reply:
hmm, I think that you are right, if context is done, `updateErr` should be a `context exceed` error and it will return ... I think that we should add an unit test demonstrating it ... there are similar tests already

Reply:
Done